src/Morphir/JsonSchema/PrettyPrinter.elm

```diff
@@ -27,15 +27,61 @@ encodeSchemaType schemaType =
             Encode.object
                 [ ( "type", Encode.string "integer" ) ]
 
-        Array st ->
-            Encode.object
-                [ ( "type", Encode.string "array" )
-                , ( "items", encodeSchemaType st )
-                ]
+        Array arrayType isUnique ->
+            case arrayType of
+                ListType itemSchemaType ->
+                    Encode.object
+                        (List.concat
+                            [ [ ( "type", Encode.string "array" )
+                              , ( "items", encodeSchemaType itemSchemaType )
+                              ]
+                            , if isUnique then
+                                [ ( "uniqueItems", Encode.bool True ) ]
 
-        String ->
-            Encode.object
-                [ ( "type", Encode.string "string" ) ]
+                              else
+                                []
+                            ]
+                        )
+
+                TupleType schemaTypes numberOfItems ->
+                    Encode.object
+                        [ ( "type", Encode.string "array" )
+                        , ( "items", Encode.bool False )
+                        , ( "prefixItems", Encode.list encodeSchemaType schemaTypes )
+                        , ( "minItems", Encode.int numberOfItems )
+                        , ( "maxItems", Encode.int numberOfItems )
+                        ]
+
+        String derivative ->
+            case derivative of
+                BasicString ->
+                    Encode.object
+                        [ ( "type", Encode.string "string" ) ]
+
+                CharString ->
+                    Encode.object
+                        [ ( "type", Encode.string "string" ) ]
+
+                DecimalString ->
+                    Encode.object
+                        [ ( "type", Encode.string "string" )
+                        ]
+
+                DateString ->
+                    Encode.object
+                        [ ( "type", Encode.string "string" )
+                        , ( "format", Encode.string "date" )
+                        ]
+
+                TimeString ->
+                    Encode.object
+                        [ ( "type", Encode.string "string" )
+                        , ( "format", Encode.string "time" )
+                        ]
+
+                MonthString ->
+                    Encode.object
+                        [ ( "type", Encode.string "string" ) ]
```

Comment:
A lot of these don't seem to have an actual format field assigned to them. This is confusing.

Reply:
Reading up on it a bit I'm not even sure if `format` is the right constraint to use here. There's also `pattern` which seems a bit more well-defined.